[ruo-x] iP

ruo-x opened the pull request:
### Bobi

> Bobi. Bobi. Bobi. Bobi. Bobi. -Bobi

Introducing Bobi, your companion to rely on for day-to-day tasks!!

What Bobi can do:

- Makes your day by greeting you
- Keeps track of all of your tasks
- Make you feel emotional by bidding you farewell

All you need to do is:

1. Download it from [here](https://www.google.com/search?gs_ssp=eJzj4tDP1TdIMy0zN2D04s5IzcnJV8jOLCmpBABOkgdF&q=hello+kitty&rlz=1C1CHBF_enSG1014SG1014&oq=hello&aqs=chrome.2.0i355i433i512j46i433i512j46i131i433i512j46i131i199i433i465i512j0i433i512j46i131i199i433i465i512j0i433i512l2j0i3j0i433i512.4050j0j7&sourceid=chrome&ie=UTF-8)
2. Double-click it
3. Add your tasks. That's it!!!

And it is **FREE!**

Features:

- [x] Add todo tasks
- [x] Add deadline tasks
- [x] Add event tasks
- [x] Mark task as done/un-done
- [x] Find for a task containing a given keyword
 
 What are you waiting for? Adopt Bobi **NOW!**

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ruo-x committed:
Greeting and exit message

ruo-x committed:
Implemented echo

ruo-x committed:
Merge pull request #1 from ruo-x/working-1

Implemented echo

ruo-x committed:
Add task and list feature

ruo-x committed:
Task status added

ruo-x committed:
New features: type of tasks, tasks saved in drive

ruo-x committed:
Used exceptions

ruo-x committed:
Delete task

ruo-x committed:
Add testing

ruo-x committed:
Update code to coding standards

ruo-x committed:
Add date and time for tasks

ruo-x committed:
Merge branch 'branch-Level-8'

ruo-x committed:
organise code for OOP

ruo-x committed:
organise into packages

ruo-x committed:
Merge remote-tracking branch 'origin/add-gradle-support' into gradle

ruo-x committed:
Set up Gradle

ruo-x committed:
JUnit tests

ruo-x committed:
create JAR file

ruo-x committed:
Add Java Doc to code

ruo-x committed:
Edit code to fit coding standards

ruo-x committed:
feature: find keyword among tasks

ruo-x committed:
Merge branch 'branch-A-JavaDoc'

ruo-x committed:
merge branch-A-JavaDoc and branch-A-CodingStandard

ruo-x committed:
merge branch-Level-9 to master

ruo-x committed:
Fix bug in FailedSearchException

ruo-x committed:
checkstyle

lshaoqin commented on src/main/java/duke/exception/FailedSearchException.java:
```diff
@@ -0,0 +1,16 @@
+
+/**
+ * FailedSearchException class inherits from DukeException class.
+ * It encapsulates the error message to be displayed when no task matches the keyword that the user has input.
```
This line wraps on my screen, so it'd be good to break it into two lines.

ruo-x replied:
Okay will do, thank you!

lshaoqin commented on src/main/java/duke/exception/InvalidEventException.java:
```diff
@@ -0,0 +1,20 @@
+    public InvalidEventException() {
+        super("Oh no! It seems like you have indicated an invalid event :/ \n"
+                + "Please follow this format:\n"
+                + "event <task name> /from <yyyy-mm-dd> <HHmm> /to <yyyy-mm-dd> <HHmm>"
```
I like how you separated the exceptions, and even provided the proper formatting of the commands in your error messages!

ruo-x replied:
Thank you :)

lshaoqin commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @return Updated string to be stored in backend.
+     */
+    public abstract String toUpdateString(int i);
```
It is unclear to me what `int i` refers to in this case, you may want consider renaming it.

ruo-x replied:
Ok! Will change the naming thanks :)

lshaoqin commented on src/main/java/duke/utility/Parser.java:
```diff
@@ -0,0 +1,182 @@
+     * @throws EmptyTaskException if user did not input a task number.
+     */
+    public static int parseActions(String input) throws EmptyTaskException {
+        String[] keyword = input.split(" ");
```
Perhaps consider `splitString` or something similar as a name for these arrays.

lshaoqin commented on src/main/java/duke/utility/Storage.java:
```diff
@@ -0,0 +1,204 @@
+     * @param task Task saved in the text file.
+     * @param i Status to update the task to. 0 for false, 1 for true.
+     */
+    public void updateTask(Task task, int i) {
```
Perhaps you can consider refactoring `int i` into a `boolean isMarked` for code clarity purposes?

lshaoqin commented on src/main/java/duke/utility/Storage.java:
```diff
@@ -0,0 +1,204 @@
+            while (sc.hasNext()) {
+                // reads text file line by line
+                String[] keyword = sc.nextLine().split("/@/");
+                switch (keyword[0]) {
+                    case "T":
+                        // line is a To-Do Task
+                        boolean tStatus = keyword[1].equals("1");
+                        list.addTask(new ToDo(tStatus, keyword[2]));
+                        break;
+                    case "D":
+                        // line is a Deadline Task
+                        boolean dStatus = keyword[1].equals("1");
+                        list.addTask(new Deadline(
+                                dStatus,
+                                keyword[2],
+                                LocalDateTime.parse(keyword[3])
+                        ));
+                        break;
+                    case "E":
+                        // line is an Event Task
+                        boolean eStatus = keyword[1].equals("1");
+                        list.addTask(new Event(
+                                eStatus,
+                                keyword[2],
+                                LocalDateTime.parse(keyword[3]),
+                                LocalDateTime.parse(keyword[4])
+                        ));
+                        break;
+                    default:
+                        System.out.println("error");
+                        break;
+                }
```
Small thing, but I believe the course conventions for switch is to have no indentation for case clauses.

lshaoqin reviewed (commented):
Great job, LGTM! I identified some small areas where styling or naming could be improved, but very clean and well formatted code overall.

mingyu-wan commented on src/main/java/duke/exception/BobiException.java:
```diff
@@ -0,0 +1,19 @@
+package duke.exception;
```
I like how you really considered alot of exceptions!

ruo-x replied:
thank you!

mingyu-wan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+     * Marks a task as done.
+     * Status of the task object is changed to true.
+     */
+    public void mark() {
```
Could use the `set` here instead for both mark and unmark for it to be even more self explanatory!

mingyu-wan commented on src/main/java/duke/Bobi.java:
```diff
@@ -0,0 +1,89 @@
+        String answer = ui.getInput();
+
+        // Listens to user commands and perform the relevant functions
+        while (true) {
```
Understandable that this is a really long chunk as theres really alot of type of input, could be simpler to divide the commands into creating tasks and handling tasks to shorten this!

mingyu-wan commented on src/main/java/duke/utility/Ui.java:
```diff
@@ -0,0 +1,119 @@
+ * @author ruo-x
+ */
+public class Ui {
+    private static final String SEPARATOR = "____________________________________________________________\n";
```
I like how you set all there `static final` so that theres no need for repeats

mingyu-wan reviewed (commented):
generally, a really clean and neat piece of code. i really like how everything is short and easy to understand. some small suggestions given but lgtm!

ruo-x committed:
Implement Bobi GUI using JavaFX and FXML

ruo-x committed:
Checkstyle changes

ruo-x committed:
Enhance code quality

ruo-x committed:
Add new Reminder feature to Bobi

This reminder feature give users an overview of their tasks that are of
a higher priority so that they can focus on those tasks first.

A reminder will pop up when user first launch the Bobi GUI. It will
remind users of any overdue tasks and tasks due this coming week as they
are usually of higher priority.

This feature is very necessary. If the user has a long list of tasks to
do, going through the list every single time will be a very big
hassle. This feature helps them stay on track as it reduces the chances
of them missing deadlines.

Yet to do:
- Javadoc comments for newly added code
- code quality check for newly added code
- consider adding a command where users can view this reminder whenever
  they want

ruo-x committed:
Merge pull request #3 from ruo-x/branch-B-Reminders

Add new Reminder feature to Bobi

ruo-x committed:
Add javadoc for reminder feature

ruo-x committed:
Merge pull request #4 from ruo-x/branch-Reminder-JavaDoc

Add javadoc for reminder feature

ruo-x committed:
Remind command and Upload Ui Screenshot

ruo-x committed:
Update storage path for multi-platforms

ruo-x committed:
Merge pull request #5 from ruo-x/branch-storage-path

Update storage path for multi-platforms

ruo-x committed:
User guide

ruo-x committed:
Merge branch 'branch-storage-path'

ruo-x committed:
Merge branch 'master' of https://github.com/ruo-x/ip
